src/loops/nested_loops/hard/q6/MainTest.java

```diff
@@ -0,0 +1,72 @@
+package loops.nested_loops.hard.q6;
+
+import global.BaseTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.TestOption;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.PlaceHolder;
+import global.variables.clauses.StringLiteral;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+public class MainTest extends BaseTest {
+    // Java
+    public Clause[] testSentence(){
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = "5";
+        return new Clause[]{
+                new StringLiteral("Enter number of lines:"),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new PlaceHolder()
+        };
+    }
+
+    public void runMain(){TryAngleButBothWaysAKAPyramid.main(new String[0]);}
+
+    static Stream<Integer> inputProvider(){
+        return Stream.of(5, 1, 0, 10, 42);
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    void printsPyramidCorrectly(int n) throws InvalidClauseException {
+        runWithInput(String.valueOf(n), clauseBuilder(n));
```

Comment:
This is missing an error message.

Reply:
I also figured here that the entire program was just formatting, I'm not even sure what an error message would be. Your pyramid was not built properly?

Reply:
I mean it is a formatting issue, but I feel like the generic message should mostly be for if they have in error in a print statement or something, whereas here we can at least specify that they printed the pattern incorrectly, and not something else